backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -129,7 +150,30 @@ internal class ObjCExportTranslatorImpl(
     }
 
     override fun translateClass(descriptor: ClassDescriptor): ObjCInterface {
-        val name = translateClassOrInterfaceName(descriptor)
+
+        val genericExportScope = genericExportScope(descriptor)
+
+        fun superClassGenerics(genericExportScope: ObjCExportScope): List<ObjCNonNullReferenceType> {
+            val parentType = computeSuperClassType(descriptor)
+            return if(parentType != null) {
+
+                parentType.arguments.map { typeProjection ->
+                    val typeParamClass = typeProjection.type.getErasedTypeClass()
+
+                    if (!typeProjection.type.isTypeParameter() && mapper.shouldBeExposed(typeParamClass)) {
+                        if (typeParamClass.isInterface)
+                            referenceProtocol(typeParamClass)
```

Comment:
What is this required for? How is it different from what `mapReferenceTypeIgnoringNullability` does?